nextflow_schema.json

```diff
@@ -1,3015 +1,2233 @@
+          "db": {
+            "type": "string",
+            "description": "Path to Kraken2 database (do not use symlinks)"
+          },
+          "classified_suffix": {
+            "type": "string",
+            "default": "classified",
+            "hidden": true
+          },
+          "unclassified_suffix": {
+            "type": "string",
+            "default": "unclassified",
+            "hidden": true
+          },
+          "report_suffix": {
+            "type": "string",
+            "default": "report",
+            "hidden": true
+          },
+          "output_suffix": {
+            "type": "string",
+            "default": "output",
+            "hidden": true
+          },
+          "save_output_fastqs": {
+            "type": "boolean",
+            "hidden": true
+          },
+          "save_reads_assignments": {
+            "type": "boolean",
+            "default": true,
+            "hidden": true
+          },
+          "run_kraken_quick": {
+            "type": "boolean",
+            "hidden": true
+          },
+          "report_tag": {
+            "type": "string",
+            "default": "KrakenReport",
+            "hidden": true
+          },
+          "tophit_level": {
+            "type": "string",
+            "default": "S",
+            "hidden": true
+          }
+        },
+        "kraken_bin": {
+          "taxonomic_level": {
+            "type": "string",
+            "default": "G",
+            "description": "Taxonomic level to bin contigs at."
+          }
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/input_output_options"
+    },
+    {
+      "$ref": "#/definitions/bakta"
+    },
+    {
+      "$ref": "#/definitions/fastp"
+    },
+    {
+      "$ref": "#/definitions/mash"
+    },
+    {
+      "$ref": "#/definitions/quast"
+    },
+    {
+      "$ref": "#/definitions/generic_options"
+    },
+    {
+      "$ref": "#/definitions/skip_options"
+    },
+    {
+      "$ref": "#/definitions/databases_and_pre_computed_files"
+    },
+    {
+      "$ref": "#/definitions/max_job_request_options"
+    },
+    {
+      "$ref": "#/definitions/institutional_config_options"
+    },
+    {
+      "$ref": "#/definitions/seqkit"
+    },
+    {
+      "$ref": "#/definitions/coveragecalculation"
+    },
+    {
+      "$ref": "#/definitions/qcreport"
+    },
+    {
+      "$ref": "#/definitions/seqtk_size"
+    },
+    {
+      "$ref": "#/definitions/medaka"
+    },
+    {
+      "$ref": "#/definitions/staramr"
+    },
+    {
+      "$ref": "#/definitions/mobrecon"
+    },
+    {
+      "$ref": "#/definitions/kat"
+    },
+    {
+      "$ref": "#/definitions/platform_options"
+    },
+    {
+      "$ref": "#/definitions/seqtk"
+    },
+    {
+      "$ref": "#/definitions/flye"
+    },
+    {
+      "$ref": "#/definitions/spades"
+    },
+    {
+      "$ref": "#/definitions/checkm"
+    },
+    {
+      "$ref": "#/definitions/kraken2"
+    }
+  ],
+  "chopper": {
```

Comment:
Parameters that aren't under a section of `definitions` need to be under a `properties` object. That is everything under here needs to be written like:

```json
  "properties": {
      "chopper": {
      // ...
      },
  // ...
}
```

Reply:
Also, each of these top-level nested parameter groups (like `chopper`) should be marked `"hidden": true`.

Reply:
fixed in latest commit! note, the hidden message appears underneath the nested block.

Reply:
Thank you. Due to the change to not include nested parameters in the schema, this is also no longer needed.